pkg/server/server.go

```diff
@@ -459,8 +463,10 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			if err := backend.Send(pkt); err != nil {
 				// TODO: retry with other backends connecting to this agent.
 				klog.ErrorS(err, "CLOSE_REQ to Backend failed", "serverID", s.serverID, "connectionID", connID)
+			} else {
+				klog.V(5).Infoln("CLOSE_REQ sent to backend", "serverID", s.serverID, "connectionID", connID)
+				return
```

Comment:
Please clarify why this is needed here, but not also in other "close" codepaths?

Also- I know we currently have single use grpc tunnel, but this change may make it harder in the future to have a multi-use tunnel.

Reply:
1. after sending close req to the backend, the work for `serveRecvFrontend` is done, right?
2. without this change, we'll send a second close req to the backend at the end of `serveRecvFrontend`. The latter one is caused by closing `recvCh` for closed connection.
Here are some log proof:
### proxy-server:
- The first time to send close req to backend

I1211 12:56:53.137935    6025 server.go:472] CLOSE_REQ sent to backend serverID 8fb992ef-fb34-475d-9fa6-de569516b226 connectionID 1
I1211 12:56:53.138643    6025 server.go:770] "Received CLOSE_RSP" serverID="8fb992ef-fb34-475d-9fa6-de569516b226" agentID="5b03efa7-411f-4a2b-9ea3-088b7235daad" connectionID=1
I1211 12:56:53.138691    6025 server.go:780] CLOSE_RSP sent to frontend connectionID 1

- skip some log
- the second time to send close rep to second, we have no log but you can tell from CLOSE_RSP

I1211 12:56:53.139887    6025 server.go:770] "Received CLOSE_RSP" serverID="8fb992ef-fb34-475d-9fa6-de569516b226" agentID="5b03efa7-411f-4a2b-9ea3-088b7235daad" connectionID=1
E1211 12:56:53.139942    6025 server.go:773] "could not get frontend client" err="can't find agentID 5b03efa7-411f-4a2b-9ea3-088b7235daad in the frontends" serverID="8fb992ef-fb34-475d-9fa6-de569516b226" agentID="5b03efa7-411f-4a2b-9ea3-088b7235daad" connectionID=1

### proxy-agent

- the first time to receive close req

I1211 12:56:53.138121    6012 client.go:456] "received CLOSE_REQ" connectionID=1
I1211 12:56:53.138161    6012 client.go:411] "close connection" connectionID=1
I1211 12:56:53.138281    6012 client.go:498] "received data from remote" bytes=0 connectionID=1
E1211 12:56:53.138312    6012 client.go:505] "connection read failure" err="read tcp 172.28.0.2:60474->172.28.0.2:8000: use of closed network connection" connectionID=1
I1211 12:56:53.138425    6012 client.go:486] "Exiting remoteToProxy" connectionID=1
I1211 12:56:53.138445    6012 client.go:524] "Exiting proxyToRemote" connectionID=1

- the second time to receive close req

I1211 12:56:53.139584    6012 client.go:375] "[tracing] recv packet" type="CLOSE_REQ"
I1211 12:56:53.139608    6012 client.go:456] "received CLOSE_REQ" connectionID=1
I1211 12:56:53.139643    6012 client.go:462] "Failed to find connection context for close" connectionID=1
E1211 13:00:53.143622    6012 client.go:371] "could not read stream" err="rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: ENHANCE_YOUR_CALM, debug data: \"too_many_pings\""

Reply:
Fortunately, this change is not a must, considering the server and agent can handle this well, but reducing these noisy error log from konnectivity will make customers feel safe.